### PR TITLE
Add stats to new Compactable classes. Fixes #1930

### LINKED ISSUE #1930: Missing MajC and Split stats in Monitor
On a cluster that I have been running for about a day, the monitor is reporting that 0 major compactions have occurred in the tserver all time history. The Split stats are screwed up as well. See screenshot.

**To Reproduce**
1. In the Monitor... Click Servers -> Tablet Servers
2. Click on the ip address link of a tserver
3. Look at the All-Time Tablet Operation Results table
image

### PR DESCRIPTION
**Pull request by milleruntime**
The stats were missing from the Monitor. See #1930

### CODE REVIEW

**Comment by milleruntime** [comment]
This will also fix the splits in the monitor.

**Review by ctubbsii — APPROVED** [review]
I don't understand the details of these changes very well, but don't see any obvious problem with them. :+1: if it fixes the splits on the monitor and gathers stats correctly.

**Comment by milleruntime** [comment]
My only concern with this fix was that it modifies the new `Compactable` interface, by adding a param to the `compact()` method.  @keith-turner any thoughts?

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java [hunk 1]
```diff
@@ -641,13 +643,18 @@ public void compact(CompactionServiceId service, CompactionJob job, RateLimiter
...
 
     StoredTabletFile metaFile = null;
+    long startTime = System.currentTimeMillis();
+    long entriesWritten = 0;
+    boolean failed = false;
+    CompactionStats stats = new CompactionStats();
     try {
 
       TabletLogger.compacting(getExtent(), job, localCompactionCfg);
+      tablet.incrementStatusMajor();
```
Comment (keith-turner) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java:653]: What does this method do?  If its nots quick to explain just let me know, I can look at the source.

**Comment by ctubbsii** [comment]
> My only concern with this fix was that it modifies the new `Compactable` interface

But, that's not public API or SPI, so it shouldn't matter.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java [hunk 2]
```diff
@@ -679,6 +688,8 @@ public void compact(CompactionServiceId service, CompactionJob job, RateLimiter
         selectedCompactionCompleted(job, jobFiles, metaFile);
       else
         selectFiles();
+
+      tablet.updateTimer(MAJOR, queuedTime, startTime, stats.getEntriesRead(), failed);
```
Comment (keith-turner) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java:692]: Could shorten the code a bit by removing failed boolean.   Please ignore this suggestion if it does not suit you. Just something I thought of when I looked at the code.

```suggestion
      tablet.updateTimer(MAJOR, queuedTime, startTime, stats.getEntriesRead(), metaFile == null);
```

**Review by keith-turner — APPROVED** [review]
> My only concern with this fix was that it modifies the new Compactable interface, by adding a param to the compact() method. @keith-turner any thoughts?

I think that change is fine.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java (hunk 1 above)
Comment (milleruntime) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java:653]: It increments the counter for ongoing major compactions. If you are wondering about synchronization, there is none. The stats are stored in a thrift object called `ActionStats`.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java (hunk 2 above)
Comment (milleruntime) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java:692]: I like it. As long as it doesn't go over the char line limit... *drum roll*

Comment (milleruntime) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java]: 98 chars!

**Pull request merged**